### PR TITLE
fix(symfony): fix swagger config to support Symfony ConfigBuilders

### PR DESCRIPTION
**Pull request by alexndlm**
| Q             | A
| ------------- | ---
| Branch?       | 3.0
| Tickets       | n/a
| License       | MIT
| Doc PR       | n/a

Please see https://symfony.com/doc/current/configuration.html#using-php-configbuilders and https://symfony.com/doc/current/components/config/definition.html#array-node-options

### CODE REVIEW

**Comment by alexndlm** [comment]
@soyuka would you please check this PR?

**Comment by soyuka** [comment]
@vincentchalamon probably has a better idea then me of the impacts of this

**Comment by vincentchalamon** [comment]
Hi @alexndlm,

Thanks for your PR. I've tested it in an API Platform project, but can't find any difference in the generated values comparing with the actual code, both dump me the following configuration:
```
array:1 [
  "foo" => array:2 [
    "name" => "Authorization"
    "type" => "header"
  ]
]
```

Can you explain why the use of `useAttributeAsKey` is necessary, and if possible add a test using it, please?

**Comment by alexndlm** [comment]
@vincentchalamon, thanks that you check this PR.

I have updated the tests and rebased PR on 2.7.

This issue is only present with [Symfony PHP ConfigBuilders](https://symfony.com/doc/current/configuration.html#using-php-configbuilders).

Now Symfony generates the following Config Class:
Снимок экрана 2022-12-19 в 11 17 08

My API Platform config:
Снимок экрана 2022-12-19 в 11 20 34


And it is not possible to specify the authorization name:
Снимок экрана 2022-12-19 в 11 18 20

With the fix in this PR, it will be the following:
Снимок экрана 2022-12-19 в 11 24 14

Снимок экрана 2022-12-19 в 11 23 12

Снимок экрана 2022-12-19 в 11 25 41

**Comment by vincentchalamon** [comment]
Thanks for the clear clarification :smiley:
Can you add a test with ConfigBuilder to reveal the bug and fix it, please?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v3.0.8
+
+### Bug fixes
+
+* [4691](https://github.com/api-platform/core/pull/4691) Symfony: Fix swagger config to support Symfony ConfigBuilders  (#4691)
```
Comment (vincentchalamon) [inline CHANGELOG.md:7]: @soyuka isn't it auto-generated, or maybe do you have another process for it?

**Review by vincentchalamon — APPROVED** [review]
Bad commitlint (easy to fix), otherwise :+1:
Good job @alexndlm!

File: CHANGELOG.md (hunk 1 above)
Comment (soyuka) [reply CHANGELOG.md:7]: yes it is auto generated please remove theses changes

Comment (alexndlm) [reply CHANGELOG.md]: Done

**Comment by vincentchalamon** [comment]
Good job @alexndlm, thanks!

**Pull request merged**